{
  "modified": "2023-08-25T16:05:33Z",
  "published": "2022-02-09T00:00:00Z",
  "schema_version": "1.5.0",
  "id": "CVE-2022-0391",
  "aliases": [
    "PSF-2022-8"
  ],
  "details": "A flaw was found in Python, specifically within the urllib.parse module. This module helps break Uniform Resource Locator (URL) strings into components. The issue involves how the urlparse method does not sanitize input and allows characters like '\\r' and '\\n' in the URL path. This flaw allows an attacker to input a crafted URL, leading to injection attacks. This flaw affects Python versions prior to 3.10.0b1, 3.9.5, 3.8.11, 3.7.11 and 3.6.14.",
  "affected": [
    {
      "package": {
        "ecosystem": "DHI",
        "name": "python",
        "purl": "pkg:dhi/python"
      },
      "severity": [
        {
          "type": "CVSS_V3",
          "score": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:H/A:N"
        }
      ],
      "ranges": [
        {
          "type": "SEMVER",
          "events": [
            {
              "introduced": "3.6.0"
            },
            {
              "fixed": "3.6.14"
            },
            {
              "introduced": "3.7.0"
            },
            {
              "fixed": "3.7.11"
            },
            {
              "introduced": "3.8.0"
            },
            {
              "fixed": "3.8.11"
            },
            {
              "introduced": "3.9.0"
            },
            {
              "fixed": "3.9.5"
            },
            {
              "fixed": "3.10.0-beta1"
            },
            {
              "introduced": "3.10.0"
            }
          ],
          "repo": "https://github.com/python/cpython"
        }
      ]
    }
  ],
  "references": [
    {
      "type": "ADVISORY",
      "url": "https://bugzilla.redhat.com/show_bug.cgi?id=2047376"
    },
    {
      "type": "REPORT",
      "url": "https://bugs.python.org/issue43882"
    }
  ]
}